lib/src/binary_codec.dart

```diff
@@ -305,5 +326,7 @@ class PostgresBinaryDecoder extends Converter<Uint8List, dynamic> {
     1184: PostgreSQLDataType.timestampWithTimezone,
     2950: PostgreSQLDataType.uuid,
     3802: PostgreSQLDataType.json,
+    46315: PostgreSQLDataType.geometry, /// TODO: Oid Changes on different databases after running `CREATE EXTENSION postgis`
```

Comment:
I think this should go into a separate PR. Do you mind if I create a stub and you can review it if it will work here?

Reply:
I'd really appreciate the assistance on that. I tried but I couldn't it right well.

Reply:
I have an idea for implementing this with a kind of pluggable interface, which not only solves this obstacle, but also it could make sure we don't necessarily bind the `postgres` package to an API that is changing under us (`package:dart_jts`). Please be a bit patient while I do the necessary preparation work, I'll keep you posted on it.

Reply:
I'd love to see how that will work. Thanks for your assistance

Reply:
This rewrite / internal refactoring seems to be a much harder problem than I first anticipated. We need to define custom encoders and decoders, and pass them through every little place that may use them. Not sure when I will be ready with that... :(

Reply:
Okay, I understand. I made some hacky solution where I send a query to read the datatypes when opening the connection,it's working well and all the other tests still work. If you'd love to check it out, and give some feedback...its on oid-connection-test branch.  
The major change is that the dynamic typeMap has to be injected into the Query object.

I've used it on a variety of images including docker images and they seem to be working okay. The only caveat is that if someone run's ``` CREATE extension custom-extension``` after connecting to the database, the typeMap will need to be updated to reflect this as the typeMap will not have the new types

Some tests are failing though.

Reply:
here's a screenshot of failing tests and passing ones since Travis CI cannot run

https://imgur.com/Zj8qqTa

Reply:
@isoos let me just close this.